docs/design/editorial_design.md

```diff
@@ -0,0 +1,354 @@
+Designing an Edit API
+=====================
+
+# Design Overview
+The industry is full of timeline management software but nearly all of them include a fundamental set of tools for manipulation. The most well known being:
+
+- Overwrite
+- Trim
+- Cut
+- Slip
+- Slide
+- Ripple
+- Roll
+- Fill (3/4 Point Edit)
+
+OpenTimelineIO, while not seeking to become a fully realized NLE, could benefit from having a simple toolkit to help automate building/modifying timelines based on time rather than index alone.
+
+## Structure
+The underlying implementation for these edit commands will be in C++, possibly under a new library called `openedit`. This has the benefit of connecting with other languages more simply (Python, Swift, C, etc.) and, hopefully, we only need expose the functionality to the bindings.
+
```

Comment:
Do you mean that you'd want a separate c++ DSO?  To sit next to OpenTime.so and OpenTimelineIO.so?  I kind of think it just belongs in the regular OpenTimelineIO library, but with its own headers.  Or are you saying you'd _just_ wrap this subset of the library and not need bindings to OpenTimelineIO?

I would probably just put them under OpenTimelineIO, possibly combined with other operations into an 'algorithms' folder:

```
- opentimelineio/
         edit/
or 
         algo/
```

Things like flatten are likely peers of this and I'd like to avoid too many namespaces for people to hunt through.

In python, I was thinking a similar arrangement:
`opentimelineio.algorithms.slip()`
etc.

Curious to hear your thoughts.

Reply:
I completely agree with this. I didn't want to make this my first move without larger consensus but this will work super well and clean up a lot of clunky code I was sketching out. As @reinecke suggested, it'll be worth pointing out that no new schemas should be allowed under `algo/`